### ISSUE TITLE
Bug: std.builtin.StackTrace cannot be written correctly into file when running the program in a tty

### ISSUE THREAD

**Issue by ikskuh** [ziglang/zig#8815]
When printing a `std.builtin.StackTrace`, it will always use the tty config determined from `stderr`, even if the file isn't actually written to the tty:

https://github.com/ziglang/zig/blob/d228d86059cf16f4b37b2853cc1323bf98d242cf/lib/std/builtin.zig#L61-L78

This will lead to zig writing ANSI escape codes into files, string buffers and other non-tty streams. Reproduction:

```zig
const std = @import("std");
pub fn main() !void
{
    const debug_info = try std.debug.getSelfDebugInfo();
    {
        var f = try std.fs.cwd().createFile("foo.txt", .{});
        defer f.close();

        try std.debug.writeCurrentStackTrace(f.writer(), debug_info, std.debug.detectTTYConfig(), @returnAddress());
    }
    std.debug.print("{s}", .{
        try std.fs.cwd().readFileAlloc(std.heap.page_allocator, "foo.txt", 50000),
    });
}
```

The genereated `foo.txt` will contain ansi escape codes, even though it's not a TTY.

I haven't seen found a proper solution for this problem yet, so this is a issue and not a PR with a solution :disappointed:

**Comment by LemonBoy** [comment]
The `format` function for `StackTrace` is quite bad, it performs a bunch of I/O and allocations using `page_allocator` behind the scenes. I'd drop this method from `StackTrace` and let the user format the stack traces as/if needed.

**Comment by MasterQ32** [comment]
Yeah, i think that would be the best solution. That can determine the required TTY config and just use `writeStackTrace`. Maybe a helper function that provides the current behaviour can be added to `std.debug`
